qmxgraph/page/api.js

```diff
@@ -358,6 +358,12 @@ graphs.Api.prototype._insertDecorationOnEdge = function _insertDecorationOnEdge
     var graph = this._graphEditor.graph;
     position = (position * 2) - 1;  // mxGraph normalizes between [-1, 1].
 
+    if (position < -1) {
```

Comment:
Could we get some think like:

```javascript
graphs.utils.clampValue = function clampValue(value, min, max) {
    if (value < min) {
        return min;
    } else if (value > max) {
        return max;
    } else {
        return value;
    )
}
```

or even better

```javascript
graphs.Api.prototype._mapDecorationPositionToMxGraph = function _mapDecorationPositionToMxGraph(position) {
  if (position < 0) {
    position = 0;
  } else if (position > 1) {
    position = 1;
  }
  return (position * 2) - 1;  // mxGraph normalizes between [-1, 1].
}
```

So we can reuse the same code in the 2 places it is called. as is today one will we will have `normalization(-1, 1) -> clamping(-1, 1)` and the other `clamping(0, 1) -> normalization(-1, 1)`.